gensim/test/test_fasttext.py

```diff
@@ -419,6 +419,8 @@ def online_sanity(self, model):
         self.assertTrue(all(['terrorism' not in l for l in others]))
         model.build_vocab(others)
         model.train(others, total_examples=model.corpus_count, epochs=model.iter)
+        # checks that `syn0` is different from `syn0_vocab`
+        self.assertFalse(np.all(np.equal(model.wv.syn0, model.wv.syn0_vocab)))
```

Comment:
This is a rather weird check. Nobody really cares if `syn0` or `syn0_vocab` are different. What we do care about is that `syn0_vocab` is incorrect because it gets overwritten by `get_vocab_word_vecs()`. A more meaningful check would be either asserting certain properties of `syn0_vocab` that should hold or checking that `syn0_vocab` is the same before and after calling `get_vocab_word_vecs()`.

Reply:
What @janpom said + check out `np.allclose` :)

Reply:
I will add another test case but let us also retain the `np.equal` check for `syn0` and `syn0_vocab`.